src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,163 @@
+package nodes.learning
+
+import scala.reflect.ClassTag
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+
+import org.apache.spark.rdd.RDD
+
+import pipelines.Logging
+import workflow.Transformer
+import utils._
+
+/**
+ * Base trait for functions that compute a kernel on a given dataset.
+ */
+trait KernelGenerator[T] {
+
+  /**
+   * Create a kernel transformer using the given dataset as one of the arguments
+   * to the kernel function. That is, if the kernel function is \phi(x, y), this
+   * binds one of the arguments using the given data.
+   *
+   * @param trainData training data matrix to be used for kernel transformation
+   * @return a kernel transformer that can be applied to train or test data
+   */
+  def fit(trainData: RDD[T]): KernelTransformer[T]
```

Comment:
Hmm.. I guess what I'm saying is that `KernelMatrix` could have both a column block access method *and* a Row access method. Certainly there's nothing that says a KernelMatrix is a row-partitioned matrix or a column partitioned matrix - but our solver currently needs the column blocked access pattern and so we need a way of telling the system "hey - for this part we want a `Seq[RDD[BDV[Double]]]`"  What's unclear to me is how having both interfaces would play with internal fitting code. I think we'd need to write a special case for this and it might not be worth it. 

Let me see if I can come up with a clean way to handle this over the weekend, otherwise we can proceed and merge without this on monday.